README
NAME
    RT::Extension::SideBySideView - SideBySide Ticket View for RT

DESCRIPTION
    Based on an original Idea from Steve Turner (MIT) and Markus Dirr
    (GreenCircle) and the ground work of Thomas Sibley (BestPractical) and
    some Ideas from BPS Wiki this AddOn will give you the option to change
    the Ticket View from original BPS View to a so called SideBySide Ticket
    View (known from wiki).

    RT users will find a new "Display History besides Metadata" option
    within the "Ticket display" section of their Preferences.

INSTALLATION
    perl Makefile.PL
    make
    make install
        May need root permissions

    Edit your /opt/rt4/etc/RT_SiteConfig.pm
        If you are using RT 4.2 or greater, add this line:

            Plugin('RT::Extension::SideBySideView');

        For RT 4.0, add this line:

            Set(@Plugins, qw(RT::Extension::SideBySideView));

        or add RT::Extension::SideBySideView to your existing @Plugins line.

    Clear your mason cache
            rm -rf /opt/rt4/var/mason_data/ob

    Restart your webserver

UPGRADING
    If you are upgrading from 0.03 or earlier, you must remove the old
    version of this extension before installing the new version by running:

        rm -rf /opt/rt4/local/plugins/RT-Extension-SideBySideView/

AUTHORS
    Torsten Brumm http://technik.picturepunxx.de/ <technik@picturepunxx.de>

    Christian Loos <cloos@netsandbox.de>

LICENCE
    This library is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.

THANKS
    Steve Turner (MIT)
    Markus Dirr (GC)
    Thomas Sibley (BPS)
    Christian Loos (NetCologne)



Changes
Revision history for RT::Extension::SideBySideView

1.00 2015-02-06
 - update Module::Install and friends
 - move config option to 'Ticket display' section
 - use JavaScript to change ticket display

0.03 2015-02-02
 - update for RT 4.0 compatibility
 - update for RT 4.2 compatibility
 - move option to 'Ticket Display as SideBySideView' section

0.02 2010-10-07
 - fix some bugs
 - using RT internal preferences system
 - Added Christian Loos as Co Author
 - rename extension to RT::Extension::SideBySideView
 - add 'LinksExtra' callback for JSGantt extension

0.01 2010-04-03
 - Initial release (as RT-Extension-SideBySideView-CustomView)


lib/RT/Extension/SideBySideView/CustomView.pm
package RT::Extension::SideBySideView::CustomView;

use 5.008003;
use strict;
use warnings;

our $VERSION = '0.02';

$RT::Config::META{'SideBySideView'} = {
    Section         => 'RT::Extension::SideBySideView',
    Overridable     => 1,
    SortOrder       => 1,
    Widget          => '/Widgets/Form/Boolean',
    WidgetArguments => {
        Description => 'SideBySideView', # loc
    },
};

1;

__END__

=head1 NAME

RT::Extension::SideBySideView

=head1 SYNOPSIS

Based on an original Idea from Steve Turner (MIT) and Markus Dirr (GreenCircle)
and the ground work of Thomas Sibley (BestPractical) and some Ideas from BPS
Wiki this AddOn will give you the option to change the Ticket View from
original BPS View to a so called SideBySide Ticket View (known from wiki).

=head1 INSTALLATION

	1. Run "perl Makefile.PL"

	2. Run "make"

	3. Run "make install" (you may need root permissions)

	4. Additionally, if you're running 3.8, you'll need to add
	"RT::Extension::SideBySideView::CustomView" to @Plugins in
	etc/RT_SiteConfig.pm.  For example:
	Set( @Plugins, qw(RT::Extension::SideBySideView::CustomView) );

	6. Clear RT's mason cache and restart the web server to make sure the
	extension is incorporated.

=head1 AUTHOR

	Torsten Brumm <tbrumm@mac.com>
	Christian Loos <cloos@netsandbox.de>

=head1 LICENCE

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=head1 THANKS

	Steve Turner (MIT)
	Markus Dirr (GC)
	Thomas Sibley (BPS)
	Christian Loos (NetCologne)

=cut


lib/RT/Extension/SideBySideView.pm
package RT::Extension::SideBySideView;

use 5.008003;
use strict;
use warnings;

our $VERSION = '1.00';

$RT::Config::META{'SideBySideView'} = {
    Section         => 'Ticket display',
    Overridable     => 1,
    Widget          => '/Widgets/Form/Boolean',
    WidgetArguments => {
        Description => 'Display History besides Metadata', # loc
        Hints       => '(' . __PACKAGE__ . ')',
    },
};

=head1 NAME

RT::Extension::SideBySideView - SideBySide Ticket View for RT

=head1 DESCRIPTION

Based on an original Idea from Steve Turner (MIT) and Markus Dirr (GreenCircle)
and the ground work of Thomas Sibley (BestPractical) and some Ideas from BPS
Wiki this AddOn will give you the option to change the Ticket View from
original BPS View to a so called SideBySide Ticket View (known from wiki).

RT users will find a new "Display History besides Metadata" option within the
"Ticket display" section of their Preferences.

=head1 INSTALLATION

=over

=item C<perl Makefile.PL>

=item C<make>

=item C<make install>

May need root permissions

=item Edit your F</opt/rt4/etc/RT_SiteConfig.pm>

If you are using RT 4.2 or greater, add this line:

    Plugin('RT::Extension::SideBySideView');

For RT 4.0, add this line:

    Set(@Plugins, qw(RT::Extension::SideBySideView));

or add C<RT::Extension::SideBySideView> to your existing C<@Plugins> line.

=item Clear your mason cache

    rm -rf /opt/rt4/var/mason_data/ob

=item Restart your webserver

=back

=head1 UPGRADING

If you are upgrading from 0.03 or earlier, you must remove the old version
of this extension before installing the new version by running:

    rm -rf /opt/rt4/local/plugins/RT-Extension-SideBySideView/

=head1 AUTHORS

Torsten Brumm http://technik.picturepunxx.de/ <tbrumm@mac.com>

Christian Loos <cloos@netsandbox.de>

=head1 LICENCE

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=head1 THANKS

=over

=item Steve Turner (MIT)

=item Markus Dirr (GC)

=item Thomas Sibley (BPS)

=item Christian Loos (NetCologne)

=back

=cut

1;
